phpBB/posting.php

```diff
@@ -44,6 +41,72 @@
 $submit = $request->is_set_post('post') && !$refresh && !$preview;
 $mode		= $request->variable('mode', '');
 
+// Only assign required URL parameters
+$forum_id = 0;
+$topic_id = 0;
+$post_id = 0;
+
+switch ($mode)
+{
+	case 'popup':
+	case 'smilies':
+		$forum_id = $request->variable('f', 0);
+	break;
+
+	case 'post':
+		$forum_id = $request->variable('f', 0);
+		if (!$forum_id)
+		{
+			trigger_error('NO_FORUM');
+		}
+	break;
+
+	case 'bump':
+	case 'reply':
+		$topic_id = $request->variable('t', 0);
+		if ($topic_id)
+		{
+			$sql = 'SELECT forum_id
+				FROM ' . TOPICS_TABLE . "
+				WHERE topic_id = $topic_id";
+			$result = $db->sql_query($sql);
+			$forum_id = (int) $db->sql_fetchfield('forum_id');
+			$db->sql_freeresult($result);
+		}
+		if (!$topic_id || !$forum_id)
```

Comment:
Please have an empty line between different `if` blocks.

Reply:
Sorry for a delayed reply, 

> I haven't gotten through all the files yet, and that probably doesn't really matter. As all these changes I will have to test locally aswell. You didn't really do yourself any favors applying the change to every single URL within phpBB, including MCP routes, posting pages, email URL's and what not, all of which are not indexed by search engines.

> As it currently stands, a few of those areas where you made the changes aswell, are not in an ideal state (the MCP for example) where there is not a straight-forward and clear approach how all authentication is handled. Cause there are a lot of places where a forum id or topic id is requested from the url and acted upon. For example `mcp_main`. So it will basically mean, we will have to test that everything works within the MCP and their respective QuickMod actions.

I believe that the weakness is in the existing design, and it is a mistake to allow a url (hacker) to choose which forum id is used for authentication, if the existing authentication logic depends on using the correct forum Id, then it is not wise to let a user (hacker) select which id is used.  If the logic does not depend on a correct forum id being supplied, then it is a redundant parameter.

> Moreover, this will mean it might be very susceptible to introducing bugs, while being based upon 3.3.x, which is once again, less than ideal.

It is my belief that removing all redundant url parameters will reduce the number of bugs in the longer term because there is less ambiguity in the behaviour of the code.

I am willing to re-base the PR to whichever code branch it is likely to be merged into.

Reply:
Somehow two unrelated comments/answers are shown here for me? @v12mike there is still an empty line missing between the above two `if` blocks, which I see everywhere below and above here. Please add this to stay consistent with coding style and let's satisfy all requirements to get this merged with v3.3.2 😃.